merkle/simple_proof.go

```diff
@@ -27,7 +27,7 @@ func SimpleProofsFromHashers(items []Hasher) (rootHash []byte, proofs []*SimpleP
 // SimpleProofsFromMap generates proofs from a map. The keys/values of the map will be used as the keys/values
 // in the underlying key-value pairs.
 // The keys are sorted before the proofs are computed.
-func SimpleProofsFromMap(m map[string]Hasher) (rootHash []byte, proofs []*SimpleProof) {
+func SimpleProofsFromMap(m map[string]Hasher) (rootHash []byte, proofs map[string]*SimpleProof, keys []string) {
```

Comment:
What is the reason for returning the keys here too? Aren't they included in the newly returned map?

Reply:
Ah OK. You need them sorted in the order as they were included and they are sorted using `.Sort()` again in https://github.com/tendermint/go-crypto/blob/9211fe9dd6cc5ad8a1052739fa674982fc914964/merkle/simple_map.go#L41